session.go

```diff
@@ -114,6 +161,112 @@ func NewSession(cfg ClusterConfig) (*Session, error) {
 		return nil, errors.New("Can't use both Authenticator and AuthProvider in cluster config.")
 	}
 
+	// Needs to be here because we need to fill in hosts and sniconfig before we do anything
+	var sniConfig *SNIConfig
+
+	if cfg.SecureConnectBundleFilename != "" {
+		// 1. Unzip the secure bundle.
+		r, err := zip.OpenReader(cfg.SecureConnectBundleFilename)
+		if err != nil {
+			return nil, err
+		}
+		defer r.Close()
+
+		// 2. Put in temp directory (the following MUST BE FILES for this to work.
+		//    a) 'cert' file
+		//    b) 'key' file
+		//    c) 'ca.crt' file
+
+		dir, err := ioutil.TempDir("", "securezip")
```

Comment:
We definitely shouldn't write to disk in gocql. It's a network client so nobody expects it will write to disk. We cannot even assume that it's possible to write to disk, that could be forbidden for example due to host app's security policy.

It is not necessary to write to disk to setup `tls.Config either`, we could just:

1. Find entries in zip for `cert`, `key` and `ca.crt` and read the contents to memory
2. Use [tls.X509KeyPair](https://golang.org/pkg/crypto/tls/#X509KeyPair) to parse the PEM key pair and put the result to `tlsConfig.Certificates`
3. Create new cert pool `x509.NewCertPool()` and add the CA certificate to it using it's `AppendCertsFromPEM`, put this to `tlsConfig.RootCAs`